docs/source/package_definition.rst

```diff
@@ -80,10 +80,15 @@ Early Binding Functions
 +++++++++++++++++++++++
 
 Early binding functions use the ``@early`` decorator. They are evaluated at *build time*, hence the
-'early' in 'early binding'. Any package attribute can be implemented as an early binding function.
+'early' in 'early binding'. By build time, we mean that they are evaluated before the resolve has happened and before
+the :ref:`build environment <the-build-environment>` is constructed. Therefore it's important to note a few important
```

Comment:
If there is no resolve, there is no environment variables.

Apart from that, it's better but I feel like we can improve the wording even more. @maxnbk @bpabel @instinct-vfx any ideas on how we could improve this paragraph?

Reply:
Not a native speaker, so i struggle with the proper wording. That said i think that the important aspect of early binding functions is that they are not only evaluated at build time but also persisted into the package. That is easy to miss and i remember struggling with some of these concepts when i first started. (It's easy to miss the declarative nature of package.py though it is python syntax)

Reply:
(?)
```
Early binding functions uses the ``@early`` decorator. They are evaluated at *build time*, hence the 'early' in 'early binding', and their definition persists into the ``package.py``. By 'build time', it is meant that they are evaluated before the resolve has occurred, and as such, before the :ref:`build environment <the-build-environment>` has been constructed. Therefore, there are some important distinctions that set early-bound functions apart from other Function Attributes.

- The :attr:`this` object only exposes package attributes. Nothing else is accessible when inside an early-bound function.
- No rez-set :doc:`environment variables <environment>` can be accessed inside an early bound function.
```

Reply:
Thanks! I like that @maxnbk. We could also add a point to mention that the returned value will be used as the package attribute in the released package. It's kind of mentioned later in the docs but a mention here would make it more explicit, like @instinct-vfx said.